src/vs/workbench/services/editor/common/editorService.ts

```diff
@@ -82,6 +82,12 @@ export interface IBaseSaveRevertAllEditorOptions {
 	 */
 	readonly includeUntitled?: boolean;
 
+	/**
+	 * Whether to include scratchpad editors.
+	 * if set to true, `includeUntitled` should also be true.
```

Comment:
this can be enforce in the type system but seemed to make creation of the parameter object more cumbersome:

```
export type IBaseSaveRevertAllEditorOptions = saveUntitledOptions & {
	readonly excludeSticky?: boolean;
};

type saveUntitledOptions = {
	includeScratchpad: false;
	includeUntitled?: boolean;
} | {
	includeScratchpad: true;
	includeUntitled: true;
};
```

Reply:
Maybe `includeUntitled` could be an object that allows to additionally specify if scratchpads are also included?